### PR TITLE
IT case with client slices and vertx

### PR DESCRIPTION
**Pull request by olenagerasimova**
Created IT case with example proxy slice, `JettyClientSlices` and Vertx.

### CODE REVIEW

**Comment by olenagerasimova** [comment]
@olegmoz could you please take a look? Something is wrong here, the result status is 500

File: src/test/java/com/artipie/http/client/jetty/JettyClientSlicesAndVertxITCase.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                this.client.https("yandex.ru").response(
+                    new RequestLine(
+                        RqMethod.GET, "/"
+                    ).toString(),
+                    Headers.EMPTY,
+                    Content.EMPTY
+                ).send(
+                    (status, rsheaders, body) -> {
+                        promise.complete(new RsFull(status, rsheaders, body));
+                        return CompletableFuture.allOf();
```
Comment (olegmoz) [inline src/test/java/com/artipie/http/client/jetty/JettyClientSlicesAndVertxITCase.java:145]: @olenagerasimova the problem is basically in this line. Once future returned by `send` method is completed HTTP client terminates `body` publisher. So when Vert.x tries to send `body` to client it is already been closed.

Here is short example how to fix this method:
```
public Response response(
            final String line,
            final Iterable<Map.Entry<String, String>> headers,
            final Publisher<ByteBuffer> pub
        ) {
            final CompletableFuture<Response> promise = new CompletableFuture<>();
            this.client.https("yandex.ru").response(
                new RequestLine(
                    RqMethod.GET, "/"
                ).toString(),
                Headers.EMPTY,
                Content.EMPTY
            ).send(
                (status, rsheaders, body) -> {
                    final CompletableFuture<Void> terminated = new CompletableFuture<>();
                    final Flowable<ByteBuffer> terminatingBody = Flowable.fromPublisher(body)
                        .doOnError(terminated::completeExceptionally)
                        .doOnTerminate(() -> terminated.complete(null));
                    promise.complete(new RsFull(status, rsheaders, terminatingBody));
                    return terminated;
                }
            );
            return new AsyncResponse(promise);
        }
```

And here is more complex version of similar code from Docker with proper error handling, see https://github.com/artipie/docker-adapter/blob/master/src/main/java/com/artipie/docker/proxy/ProxyBlob.java#L102

I know this looks complicated and needs to simplified somehow. Maybe we can create some class to hide all this boilerplate code

**Comment by olenagerasimova** [comment]
@olegmoz thanks for the explanation, I've used your example in the test.

**Review by olegmoz — APPROVED** [review]
@olenagerasimova thanks, merging

**Pull request merged**